FSW/main.cpp

```diff
@@ -0,0 +1,44 @@
+/* main.cpp is the main script for the NanoSAM II FSW 
+ * Usage:
+ *  run this script through Teensyduino to power the NS2 payload
+ * 
+ * Additional files needed for compilation:
+ *  memUtil.hpp
+ *  memUtil.cpp
+ *
+ */
+
+/* - - - - - - Includes - - - - - - */
+// C++ libraries
+
+// Other libraries
+
+// NS2 headers
+#include "headers/memUtil.hpp"
```

Comment:
I'm unfamiliar with how this inclusions work in C++. What would make sense to me is that you require the main file (memUtil.cpp) which in turn requires the header file. Can you explain this to me?

Reply:
the key here is that when compiling the program, we run a line like this:

`g++ script1.cpp script2.cpp ... scriptN.cpp`

This compiles all of the files (so `main.cpp` and each util script in our case) into a single executable. The header files being included ensures that all functions being called are defined properly and do not throw a compilation error. Putting each of these declarations above `main()` in `main.cpp` is tedious for large projects, which is why we use header files.

Reply:
Also the Arduino IDE does this compilation for us automatically if we provide it with a file structure it understands as far as I can tell. I will want to test this though ASAP

Reply:
Interesting good to know. I imagine that command will get long then as we start to have a ton of different functions. Is there anyway that is avoided, I.E with a Makefile or something?

Reply:
yeah we could avoid it with a makefile if we wanted to. I don't anticipate having more than 4 or 5 util scripts in FSW, but I will add this as a topic for tuesday's meeting.